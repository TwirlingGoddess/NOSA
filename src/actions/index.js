export const addImages = (images) => {
  return {
    type: 'ADD_IMAGES',
    images
  };
};

export const addNosa = (pic) => {
  return {
    type: 'ADD_PIC',
    pic
  }
}