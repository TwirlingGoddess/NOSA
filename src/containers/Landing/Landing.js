import React, { Component } from 'react';
import { fetchNasaPics } from '../../helpers/fetchCalls.js';

export default class Landing extends Component {
 
 componentDidMount = () => {
  fetchNasaPics()
 }

 changeToPic = () => {
  console.log('changing from landing to judging Pic')
 }

 render() {
  return(
    <div>
      <button type='submit' onSubmit={this.changeToPic}>PICS</button>
      <button>ARTICLES</button>
      <button>MERCH</button>
    </div>
  )
 }
}