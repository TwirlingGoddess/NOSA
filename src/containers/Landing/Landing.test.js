import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Landing, mapStateToProps, mapDispatchToProps } from './Landing';
import { addImages } from '../../actions';

describe('Landing container', () => {
  let wrapper;

  describe('Landing component', () => {
    it('should match the snapshot', () => {
      wrapper = shallow(<Landing />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return an object with the images array', () => {
      const mockState = {
        images: [{ url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: '' }],
        clicked: true,
      };
      const expected = {
        images: [{ url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: '' }],
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it('should calls dispatch when the addImages funtion is invoked', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addImages([{ url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: '' }]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addImages([{ url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: '' }]);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
