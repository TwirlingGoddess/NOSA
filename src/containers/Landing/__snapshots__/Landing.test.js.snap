// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing container Landing component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/prints"
>
          <button>
                    PRINTS
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/pics"
>
          <button
                    type="submit"
          >
                    PICS
          </button>
</NavLink>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/articles"
>
          <button>
                    ARTICLES
          </button>
</NavLink>,
      ],
      "className": "Landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button>
            PRINTS
</button>,
          "to": "/prints",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PRINTS",
          },
          "ref": null,
          "rendered": "PRINTS",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button
            type="submit"
>
            PICS
</button>,
          "to": "/pics",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PICS",
            "type": "submit",
          },
          "ref": null,
          "rendered": "PICS",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <button>
            ARTICLES
</button>,
          "to": "/articles",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ARTICLES",
          },
          "ref": null,
          "rendered": "ARTICLES",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/prints"
>
            <button>
                        PRINTS
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/pics"
>
            <button
                        type="submit"
            >
                        PICS
            </button>
</NavLink>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/articles"
>
            <button>
                        ARTICLES
            </button>
</NavLink>,
        ],
        "className": "Landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button>
              PRINTS
</button>,
            "to": "/prints",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PRINTS",
            },
            "ref": null,
            "rendered": "PRINTS",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button
              type="submit"
>
              PICS
</button>,
            "to": "/pics",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PICS",
              "type": "submit",
            },
            "ref": null,
            "rendered": "PICS",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <button>
              ARTICLES
</button>,
            "to": "/articles",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ARTICLES",
            },
            "ref": null,
            "rendered": "ARTICLES",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
